iree/tools/compilation.bzl

```
@@ -9,17 +9,40 @@
 load("//build_tools/embed_data:build_defs.bzl", "c_embed_data")
 
 # TODO(benvanik): port to a full starlark rule, document, etc.
+
 def iree_bytecode_module(
         name,
         src,
+        module = None,
         flags = ["-iree-mlir-to-vm-bytecode-module"],
         translate_tool = "//iree/tools:iree-translate",
         embedded_linker_tool = "@llvm-project//lld:lld",
         opt_tool = "//iree/tools:iree-opt",
         opt_flags = [],
         c_identifier = "",
         **kwargs):
+    """Builds an IREE bytecode module.
+
+    Args:
+        name: Name of the target
+        src: mlir source file to be compiled to an IREE module.
+        flags: additional flags to pass to the compiler. Bytecode
+            translation and backend flags are passed automatically.
+        translate_tool: the compiler to use to generate the module.
+            Defaults to iree-translate.
+        embedded_linker_tool: the embedded linker to use.
+            Defaults to the lld from the llvm-project directory.
+        opt_tool: Defaulting to iree-opt. Tool used to preprocess the source file
+            if opt_flags is specified.
+        opt_flags: If specified, source files are preprocessed with opt_tool with
+            these flags.
+        module: Optional. Specifies the  path to use for the enerated IREE module (.vmfb).
+        c_identifier: Optional. Enables embedding the module as C data.
+        **kwargs: any additional attributes to pass to the underlying tests and test suite.
```

No test or test suite here. Also please forward the kwargs to the opt genrule (missed that in previous review)

```suggestion
        **kwargs: any additional attributes to pass to the underlying rules.
```